app/models/pawn.rb

```diff
@@ -1,3 +1,33 @@
 class Pawn < Piece
 
+  def is_valid?(new_x, new_y)
+    return false unless super   
+    return false unless allowed_move?(new_x, new_y)
+    return false if opponent_here?(new_x, new_y) and not allowed_capture?(new_x, new_y)
+    return true
+  end
+
+  def allowed_capture?(new_x, new_y)
+    white ? allowed_diagonal = 1 : allowed_diagonal = -1
+    return opponent_here?(new_x, new_y) && (new_y - y_coordinate) == allowed_diagonal && (new_x - x_coordinate).abs == 1 
+  end
+
+  def allowed_move?(new_x, new_y)
+    white ? allowed_moves = [1] : allowed_moves = [-1]
+    allowed_moves.push(allowed_moves[0]*2) if first_move?()

```

Comment:
Good idea @eyoshimoriyama ! My concern is that it is really tightly coupled with the position of white/black pieces. (1, -1 move). Is there a way we can solve the `first_move?` in a more generic way maybe? Maybe using a flag `moved` or similar that we can add on pieces?

Reply:
Ok I think that makes sense. Would this mean creating at @moved variable that is set at false and gets changed to true when a piece is moved? Or alternatively I think can check if update_at == created_at to see if it's the first move.